dds/DCPS/TypeSupportImpl.h

```diff
@@ -173,9 +169,34 @@ class TypeSupportImpl_T : public TypeSupportImpl {
   {
     return MarshalTraitsType::key_only_serialized_size_bound(encoding);
   }
-};
 
-const char* kind_to_string(const XTypes::EquivalenceKind ek);
+#ifndef OPENDDS_SAFETY_PROFILE
+  DDS::DynamicType_ptr get_type()
+  {
+    if (!type_lookup_service_) {
+      type_lookup_service_ = make_rch<XTypes::TypeLookupService>();
+      add_types(type_lookup_service_);
+      populate_dependencies(type_lookup_service_);
+    }
+
+    const XTypes::TypeIdentifier& cti = getCompleteTypeIdentifier();
+    const XTypes::TypeMap& ctm = getCompleteTypeMap();
+    const XTypes::TypeIdentifier& mti = getMinimalTypeIdentifier();
+    const XTypes::TypeMap& mtm = getMinimalTypeMap();
+    XTypes::DynamicTypeImpl* dt = dynamic_cast<XTypes::DynamicTypeImpl*>(
+      type_lookup_service_->type_identifier_to_dynamic(cti, GUID_UNKNOWN));
+    dt->set_complete_type_identifier(cti);
+    dt->set_complete_type_map(ctm);
+    dt->set_minimal_type_identifier(mti);
+    dt->set_minimal_type_map(mtm);
+
+    return dt;
+  }
```

Comment:
I don't think I can directly because this class is a template. If we're not following the `*_T.cpp` pattern that uses `ACE_TEMPLATES_REQUIRE_SOURCE`, then I think I'd have to make an intermediate parent class between this class and `TypeSupportImpl` to hold `type_lookup_service_` and then I can put this method in the cpp.

Reply:
I didn't notice it was a template since that didn't show up in the diff.  If nothing depends on the template argument than moving it to a base class (could that be the existing `TypeSupportImpl`?) would be good.

Reply:
Yea, I wrote the intermediate class, but just realized it's a little weird, so came back to say I'm going to put it in `TypeSupportImpl`.